### ISSUE TITLE
Provide DeerLab through Anaconda Repo

### ISSUE THREAD

**Issue by laenan8466** [JeschkeLab/DeerLab#12]
Hey,
first congrats moving to python. 😍 

From my experience, many researchers use Anaconda (https://www.anaconda.com/) to organize their python packages. So it could be helpful to provide DeerLab via Anaconda as well.
https://docs.anaconda.com/anaconda-cloud/user-guide/getting-started/

If not an option for you feel free to close this issue, I would appreciate consideration, as it would simplify the workflow, especially for python-n00bs. 😉 

Cheers,
Dennis

**Comment by stestoll** [comment]
Makes sense to me - unless it's too much additional trouble.

**Comment by luisfabib** [comment]
Agreed, this would make sense.

I am not too familiar with Anaconda and the conda manager, so I will need to look at how conda manages the BLAS/LAPACK binaries of numpy/scipy on each OS.

**Comment by mtessmer** [comment]
Last I checked conda defaults to mkl numpy/scipy on all operating systems. Not sure about cvxopt but I know cvxopt is available.

**Comment by stestoll** [comment]
I think `pip` can be used from inside `conda`, so it should not be necessary to provide DeerLab via Anaconda.

**Comment by laenan8466** [comment]
I would agree (in principle), but it is not recommended to use `pip` in a `conda` environment. But yes, it is possible to use `pip` with Anaconda.
https://www.anaconda.com/blog/using-pip-in-a-conda-environment

**Comment by luisfabib** [comment]
For now, DeerLab will only be provided through `pip`. 

Maybe in a future release, this will change, but for now, this would introduce too many maintenance issues in the short-term future which should be focused on more critical development of the software.

**Comment by luisfabib** [comment]
After some preliminary trials to distribute DeerLab through `conda` and the Anaconda repo, it seems that the current `pip` setup is incompatible with the `conda` system. 

Making this compatible would require either a major hack to make it `conda`-compatible, or remake the whole `pip` installation. The issue is re-opened as we still consider this, but cannot implement it at the current time.

**Comment by laenan8466** [comment]
For reference, how to use DeerLab with Anaconda (for everyone else who stumbles over this problem). This is from my memory, so @luisfabib can comment if this way causes problems as well.

## Installation
Open your git console.
Choose a local directory, where the files should be stored.
```
cd /local/directory/
git clone git@github.com:JeschkeLab/DeerLab.git
cd DeerLab
git checkout v0.12.2
```
Replace the `v0.12.2` with the current latest version found on the [releases page](https://github.com/JeschkeLab/DeerLab/releases).

Open the `Anaconda Prompt` as admin and navigate to the directory, then activate your environment (optional)
Installation will be made with symlink, so updates in the folder will be availible after a kernel restart automatically
```
cd /local/directory/DeerLab
# activate installation-env   # change to your environement, optional
pip install -e .            # -e statement will establish a symlink
```

## Update
Open a git console and go to the local directory, git pull and checkout new version.

``` 
cd /local/directory/DeerLab
git pull
git checkout $NEW_VERSION_TAG
```

Replace `$NEW_VERSION_TAG` with the newest version from [releases page](https://github.com/JeschkeLab/DeerLab/releases).

**Comment by luisfabib** [comment]
Thanks for the instructions! I have updated the documentation installation instructions to include this workaround for Anaconda users (see #155).

**Comment by laenan8466** [comment]
Just realizing I made a minor error:
The git clone call should use the https link, as most users wouldn‘t have a git account with stored ssh-key.
`git clone https://github.com/JeschkeLab/DeerLab.git`
should do the magic.

**Comment by stestoll** [comment]
Closed by #157
